### PR TITLE
#7 Create Entity Model Classes

### PR DESCRIPTION
**Pull request by noor848**

### CODE REVIEW

File: Conduit/Conduit/bin/Debug/net6.0/appsettings.Development.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (hayatzsala) [inline Conduit/Conduit/bin/Debug/net6.0/appsettings.Development.json:1]: @noor848 again, the bin folder should be in .gitignore

File: Conduit/Conduit/bin/Debug/net6.0/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (hayatzsala) [inline Conduit/Conduit/bin/Debug/net6.0/appsettings.json:1]: .gitignore

File: Conduit/Conduit.Db/Article.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Conduit.Db
+{
+    public class Article
+    {
+        public int ArticleId { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Article.cs]: @noor848 why not Guid or long? not recommended to use int as id

File: Conduit/Conduit.Db/Comment.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Conduit.Db
+{
+    public class Comment
+    {
+        public int CommentId { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Comment.cs]: @noor848 same as articleId

File: Conduit/Conduit.Db/Article.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public string Image { get; set; }
+        public ICollection<Comment> Comments { get; set; }
+        public ICollection<Favourite> Favourites { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Article.cs]: @noor848 actually there is no need for these collection, we replace it by using separate Table/mode

File: Conduit/Conduit.Db/ConduitContext.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+namespace Conduit.Db
+{
+    public  class ConduitContext: DbContext
+    {
+        public ConduitContext(DbContextOptions<ConduitContext> options) : base(options)
+        {
+
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/ConduitContext.cs:17]: @noor848 do you create the DB from SQL files? if yes , can you attached them in folder please?
if No, you can here declare the key for each table (primary/foregin)

File: Conduit/Conduit.Db/Favourite.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Conduit.Db
+{
+    public class Favourite
+    {
+        public int FavouriteId { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Favourite.cs]: @noor848 all ids here same as ArticleId comment

File: Conduit/Conduit.Db/Follower.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Conduit.Db
+{
+    public class Follower
+    {
+        [DatabaseGenerated(DatabaseGeneratedOption.None)]
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Follower.cs]: no need for this attribute, actually when you create new object just assign new Guid() to the DB model

File: Conduit/Conduit.Db/User.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class User
+    {
+        public Guid UserId { get; set; }
+        public string  UserName { get; set; }
+        public string  Email { get; set; }
+        public string  Password { get; set; }
+        public string  Bio { get; set; }
+        public int  Age { get; set; }
+        public ICollection<Comment> Comments{ get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/User.cs]: @noor848 hmmm, thinking deeply here do we have any relation with comment? NO, so no need for this, if you need to add relation so you have to add column to comment table "CreatedBy" and be a foreign key to user table

File: Conduit/Conduit.Db/User.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        public ICollection<Favourite> Favourites{ get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/User.cs]: @noor848 as a comment on Article

File: Conduit/Conduit.Db/User.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        public ICollection<Article> Articles{ get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/User.cs]: same as the comment of "comment createdBy"

File: Conduit/Conduit/Program.cs [hunk 12]
```diff
@@ -6,6 +17,33 @@
...
+});
+builder.Services.AddScoped<IAuthService,AuthService>();
+
+builder.Services.AddScoped<IpasswordHasher,BycryptPasswordHasher>();
+
+builder.Services.AddTransient<IUserRepositry,UserRepositry>();
+builder.Services.AddAutoMapper(AppDomain.CurrentDomain.GetAssemblies());
+builder.Services.Configure<Jwt>(builder.Configuration.GetSection("JWT"));
+
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme).AddJwtBearer(options =>
```
Comment (hayatzsala) [inline Conduit/Conduit/Program.cs]: hmmmm, why do I see these changes here? if you checkout the branch from the main so I shouldn't see the changes here,and you can isolate it with DB changes.

File: Conduit/Conduit.Db/Article.cs (hunk 3 above)
Comment (noor848) [reply Conduit/Conduit.Db/Article.cs]: i did it for user and follower ill  change it for the rest

File: Conduit/Conduit/bin/Debug/net6.0/appsettings.Development.json (hunk 1 above)
Comment (noor848) [reply Conduit/Conduit/bin/Debug/net6.0/appsettings.Development.json:1]: don in the second branch

**Comment by noor848** [comment]
Conduit #7

File: Conduit/Conduit.Db/Favourite.cs [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Conduit.Db
+{
+    public class Favourite
+    {
+        public Guid FavouriteId { get; set; }
+        public Guid UserId { get; set; }
+        public int ArticleId { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Favourite.cs]: AtricleId is Guid not int, don't forget to update it here

File: Conduit/Conduit.Db/User.cs [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class User
+    {
+        public Guid UserId { get; set; }
+        public string  UserName { get; set; }
+        public string  Email { get; set; }
+        public string  Password { get; set; }
+        public string  Bio { get; set; }
+        public int  Age { get; set; }
+       /// public ICollection<Comment> Comments{ get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/User.cs]: clean code : remove unuseful comments

**Review by hayatzsala — APPROVED** [review]
Approved, 2 tiny comments, if you can update it before merge please

**Review by hayatzsala — APPROVED** [review]

**Review by hayatzsala — APPROVED** [review]

**Pull request merged**